### PR TITLE
Add java/OlaClient.getUniverseList()

### PR DESCRIPTION
**Pull request by hepiyellow**
Add missing Java OlaClient method for fetching all universes.
The RPC already existed, just added the Java method, which is similar to the existing java/OlaClient.getUniverseInfo(int universe), just without the `universe` argument.

BTW: I have an issue running ola's top-level `make` on my machine so i didn't do it. I only run the `java/make ola.jar'.

### CODE REVIEW

File: java/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+src/main/java/ola/proto/
```
Comment (peternewman) [inline java/.gitignore]: Can you put this all in our central .gitignore in root please.

File: java/src/main/java/ola/OlaClient.java [hunk 2]
```diff
@@ -180,6 +180,18 @@ public UniverseInfoReply getUniverseInfo(int universe) {
 
 
     /**
+     * Get list of all universe infos.
```
Comment (peternewman) [inline java/src/main/java/ola/OlaClient.java]: This would probably be better written as "Get a list of info about all universes"

**Comment by peternewman** [comment]
What's the error you get running the top level Make command?

File: java/ChangeLog [hunk 3]
```diff
@@ -0,0 +1,2 @@
+04/22/2017 Erez Makavy <hepi.yellow@gmail.com>
```
Comment (peternewman) [inline java/ChangeLog]: If we're going to have a ChangeLog (and I'm personally undecided), can you add some info about v0.0.1 too please.

**Comment by hepiyellow** [comment]
I am getting an error when doing `sudo make install` (after `make` and `make check` worked).
This is the error:
```
$ sudo make install
Password:
/Applications/Xcode.app/Contents/Developer/usr/bin/make  install-recursive
Making install in java
mvn package
Exception in thread "main" java.lang.UnsupportedClassVersionError: org/apache/maven/cli/MavenCli : Unsupported major.minor version 51.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClassCond(ClassLoader.java:637)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:621)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
	at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClassFromSelf(ClassRealm.java:401)
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:42)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:254)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	at org.codehaus.plexus.classworlds.launcher.Launcher.getMainClass(Launcher.java:144)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:266)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
make[2]: *** [ola.jar] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2
```


I don't think it is related to my change?
Anyways, any idea how to solve it?

File: java/src/main/java/ola/OlaClient.java (hunk 2 above)
Comment (hepiyellow) [inline java/src/main/java/ola/OlaClient.java]: done

File: java/.gitignore (hunk 1 above)
Comment (hepiyellow) [inline java/.gitignore]: done

File: java/ChangeLog (hunk 3 above)
Comment (hepiyellow) [inline java/ChangeLog]: done

**Comment by peternewman** [comment]
Can you resync and resolve conflicts please @hepiyellow , I've just merged some changes so we can test the Java stuff via Travis.

**Comment by peternewman** [comment]
I'm not sure about the error @hepiyellow . Do you want to try a checkout of the previous code just to confirm it's not you (I suspect it isn't too).

**Comment by peternewman** [comment]
General discussion point, should Java API versions track the OLA versions, or stay independent?

**Comment by peternewman** [comment]
Hi @hepiyellow . Can you fix the build issue please, then we can look at merging this. Thanks

**Comment by hepiyellow** [comment]
I updated my branch to fix the cleanup of any version of ola-java-client.
Excuse me for the newbie question, should I do anything after updating the PR brand on my fork? like requesting a PR merge or something? or it is automatic whenever I update the branch?

**Comment by peternewman** [comment]
Your changes have appeared fine thanks @hepiyellow . I'll try and take a look soon.

File: java/Makefile.am [hunk 4]
```diff
@@ -25,9 +25,9 @@ CLEANFILES = generated-src/main/java/ola/proto/Ola.java \
              classes/ola/*.class \
              test-classes/ola/*.class \
              maven-archiver/pom.properties \
-	     surefire-reports/ola.*.txt \
-	     surefire-reports/TEST-*.xml \
-             ola-java-client-0.0.1.jar
+             surefire-reports/ola.*.txt \
```
Comment (peternewman) [inline java/Makefile.am]: Can you fix the whitespace on these three lines, it should all be spaces, not tabs, so they're aligned equally with the lines above and below.

Comment (peternewman) [inline java/Makefile.am]: Done.

**Comment by peternewman** [comment]
@nomis52 have you got some time to review this before we merge it?

**Pull request merged**